Remove deprecated o2::tpc::Cluster data format

davidrohr opened the pull request:
Finally a PR to remove the old version of the TPC cluster format as decided ages ago @wiechula @matthiasrichter @sawenzel @shahor02 @a-mathis 

I also removed some related utilities: the box clusterer, dEdx via truncated mean (the new TPC track now has dEdx), and some utility macros.
Some macros still seemed useful but have not been adapted yet, so I just commented out so they will not fail in the macro tests. This should be followed up.

Please let me know if you agree with this.

shahor02 commented:
@davidrohr: looks fine, the ``Detectors/TPC/reconstruction/macro/runRawClusterFinder.C`` should also be removed.

a-mathis commented:
@davidrohr Thanks, looks fine. Maybe consider to remove the remaining macros, most of them are outdated (and rather trivial) anyways.

davidrohr commented:
@a-mathis : The remaining macros are Detectors/TPC/reconstruction/macro/dEdxRes.C, Detectors/TPC/reconstruction/macro/testClustererData.C, and Detectors/TPC/reconstruction/macro/testTracks.C.
Shall I remove all of them?

a-mathis commented:
@davidrohr Maybe @wiechula can comment on the Detectors/TPC/reconstruction/macro/testClustererData.C, but the rest are rather trivial macros and could be to my opinion remove

wiechula commented:
Hi @davidrohr ,
thanks a lot for the work. From my side all macro can be removed.
Cheers,
Jens

matthiasrichter commented:
I suggest to wait for the CI to finish and then merge

davidrohr commented:
Yes, I'll do, but probably the MacOS C.I. will fail because the is currently a problem with the CCDB test.
In principle, we could merge anyway. This PR passed the C.I. yesterday, and I just rebased it and removed a macro now.

davidrohr commented:
@sawenzel @matthiasrichter : Can we merge this? It only fails the ccdb test, which is currently failing everywhere, and actually it passed the test already yesterday and was all green.
